packages/ui/components/MediaControls/MediaButtons.tsx

```diff
@@ -0,0 +1,139 @@
+import { FunctionComponent, useState, useEffect, useRef } from 'react'
+import cn from 'classnames'
+import { BsPlus } from "@react-icons/all-files/bs/BsPlus"
+import { BsDash } from "@react-icons/all-files/bs/BsDash"
+import { BsArrowsFullscreen } from "@react-icons/all-files/bs/BsArrowsFullscreen"
+
+import { BsFillVolumeMuteFill } from "@react-icons/all-files/bs/BsFillVolumeMuteFill"
+import { BsFillVolumeDownFill } from "@react-icons/all-files/bs/BsFillVolumeDownFill"
+import { BsFillVolumeUpFill } from "@react-icons/all-files/bs/BsFillVolumeUpFill"
+import { BsPlay } from "@react-icons/all-files/bs/BsPlay"
+
+type Props = {
+  play: () => void
+  pause: () => void
+  isPlaying: boolean
+  isVideo: boolean
+  addTen: () => void
+  subTen: () => void
+  requestFullscreen: () => void
+  volumePercentage: number
+  muted: boolean  
+  toggleMuted: () => void
+  toggleVolume: (clientRectLeft: number, clientX: number) => void
+}
+
+export const MediaButtons: FunctionComponent<Props> = ({
+  play,
+  pause,
+  isVideo,
+  isPlaying,
+  addTen,
+  subTen,
+  requestFullscreen,
+  volumePercentage,
+  muted,
+  toggleMuted,
+  toggleVolume
+}) => {
+  
+  const volumeRef = useRef<HTMLDivElement | null>(null)
+  const [volumeIcon, handleVolumeIcon] = useState(null)
+
+  const getVolumeIcon = () => {
```

Comment:
This should be a separate component that receives muted and volumePercentage as props. That way you also avoid storing the component as if it were state.